application/client/src/app/service/session/dependencies/search/state.ts

```diff
@@ -1,26 +1,29 @@
-import { FilterRequest } from './filters/request';
 import { IFilter } from '@platform/types/filter';
 import { Subjects, Subject } from '@platform/env/subscription';
 import { Search } from '@service/session/dependencies/search';
 
 import * as obj from '@platform/env/obj';
 
+export interface IFinish {
+    found: number;
+    error?: string;
+}
+
 export class State {
     public readonly subjects: Subjects<{
         active: Subject<IFilter | undefined>;
-        collection: Subject<FilterRequest[]>;
-        start: Subject<void>;
-        finish: Subject<number>;
+        start: Subject<IFilter[]>;
+        finish: Subject<IFinish>;
     }> = new Subjects({
         active: new Subject<IFilter | undefined>(),
-        collection: new Subject<FilterRequest[]>(),
-        start: new Subject<void>(),
-        finish: new Subject<number>(),
+        start: new Subject<IFilter[]>(),
+        finish: new Subject<IFinish>(),
     });
 
     private _search: Search;
     private _active: IFilter | undefined;
     private _hash: string | undefined;
+    private _searching!: boolean;
```

Comment:
no need to have `undefined` as possible value... state is created as soon as session is created. On new session we don't have search in progress. Well, it should be just
```
private _searching: boolean = false;
```

Reply:
Makes sense, yeah. That'd also get rid of a few extra lines in code.